R/derive_vars_computed.R

```diff
@@ -0,0 +1,194 @@
+#'   codes. For example, `constant_parameters = exprs(WEIGHT, HGHT = VSTESTCD ==
+#'   "HEIGHT")` uses the parameter code `"WEIGHT"` and creates a temporary
+#'   parameter code `"HGHT"`.
+#'
+#'   *Permitted Values:* A character vector of `PARAMCD` values or a list of expressions
+#'
+#' @param constant_by_vars By variables for constant parameters
+#'
+#'   The constant parameters (parameters that are measured only once) are merged
+#'   to the other parameters using the specified variables.
+#'   (Refer to the Example)
+#'
+#'   *Permitted Values:* list of variables
+#'
+#'
+#' @param new_vars Name of the newly created variables
+#'
+#'   The specified variables are set to the specified values. The values of
+#'   variables of the parameters specified by `parameters` can be accessed using
+#'   `<variable name>.<parameter code>`. For example
+#'   ```
+#'   exprs(
+#'     BMIBL = (AVAL.WEIGHT / (AVAL.HEIGHT/100)^2)
+#'   )
+#'   ```
+#'   defines the value for the new variable.
+#'
+#'   Variable names in the expression must not contain more than one dot.
+#'
+#'   *Permitted Values:* List of variable-value pairs
+#'
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` argument), the values of the new variables (`new_vars`) are
+#'   computed based on the parameters in the additional dataset
+#'   (`dataset_add`) and then the new variables are merged to the input
+#'   dataset (`dataset`).
+#'
+#' @return The input dataset with the new variables added.
+#'
+#' @family der_gen
+#'
+#' @keywords der_gen
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr)
+#'
+#' # Example 1: Derive BMIBL
+#' adsl <- tribble(
+#'   ~STUDYID,   ~USUBJID, ~AGE,   ~AGEU,
+#'   "PILOT01", "01-1302",   61, "YEARS",
+#'   "PILOT01", "17-1344",   64, "YEARS"
+#' )
+#'
+#' advs <- tribble(
+#'   ~STUDYID, ~USUBJID, ~PARAMCD, ~PARAM, ~VISIT, ~AVAL, ~AVALU, ~ABLFL,
+#'   "PILOT01", "01-1302", "HEIGHT", "Height (cm)", "SCREENING", 177.8, "cm", "Y",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "SCREENING", 81.19, "kg", "N",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "BASELINE", 82.1, "kg", "Y",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "WEEK 2", 81.19, "kg", "N",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "WEEK 4", 82.56, "kg", "N",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "WEEK 6", 80.74, "kg", "N",
+#'   "PILOT01", "17-1344", "HEIGHT", "Height (cm)", "SCREENING", 163.5, "cm", "Y",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "SCREENING", 58.06, "kg", "N",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "BASELINE", 58.06, "kg", "Y",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "WEEK 2", 58.97, "kg", "N",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "WEEK 4", 57.97, "kg", "N",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "WEEK 6", 58.97, "kg", "N"
+#' )
```

Comment:
@bundfussr should this be `VS` rather than `ADVS`.  I would think we would create the `BMIBL` from `VS` .